src/view/screens/Profile.tsx

```diff
@@ -84,12 +85,18 @@ export function ProfileScreen({route}: Props) {
     }
   }, [profile?.viewer?.blockedBy, resolvedDid])
 
-  if (isLoadingDid || isLoadingProfile || !moderationOpts) {
+  // Most pushes will happen here, since we will have only placeholder data
+  if (isLoadingDid || isLoadingProfile || isPlaceholderProfile) {
+    const moderation =
+      profile && moderationOpts
+        ? moderateProfile(profile, moderationOpts)
+        : null
+
     return (
       <CenteredView>
         <ProfileHeader
-          profile={null}
-          moderation={null}
+          profile={profile}
+          moderation={moderation}
```

Comment:
Instead of waiting for `moderationOpts` to render, just pass null if we don't get it and let `ProfileHeader` determine whether it's ready or not.